File: 0055-Jump-Game/lc0055.cc
// C++ solution.

class Solution {
public:
  bool canJump(vector<int>& nums) {
    int expect = 0;
    for (int i = 0; i < nums.size(); i++) {
      if (i > expect) { return false; }
      expect = max(expect, i+nums[i]);
    }
    return true;
  }
};

File: 0055-Jump-Game/lc0055.rs
// Rust solution.

impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let mut expect = 0_i32;
        for i in 0..nums.len() {
            if (i > expect as usize) { return false; }
            expect = std::cmp::max(expect, i as i32 + nums[i]);
        }
        true
    }
}
